packages/components/src/util/typography.tsx

```diff
@@ -0,0 +1,88 @@
+import React, { ReactNode } from "react";
+
+import classNames from "classnames/bind";
+import styles from "./typography.module.css";
+
+const cx = classNames.bind(styles);
+
+export type TypographySize =
+  | "h1"
+  | "h2"
+  | "h3"
+  | "h4"
+  | "h5"
+  | "body"
+  | "sm"
+  | "xs"
+  | "caption"
+  | "overline";
+
+export type TypographyColor =
+  | "alert"
+  | "base"
+  | "brand"
+  | "info"
+  | "neutral"
+  | "success"
+  | "warning"
+  | "white";
+
+export type TypographyProps<IComponent extends React.ElementType> = {
```

Comment:
@ahuth not blocking but would appreciate your review on the typescript because I certainly know less than either of you

Reply:
Looks pretty good. My only suggestion is to drop the 'I' prefix from the type param. I don't believe it's needed (although let me know if it is), and we shouldn't get in the habit of adding it.

```suggestion
export type TypographyProps<Component extends React.ElementType> = {
```

or even

```suggestion
export type TypographyProps<T extends React.ElementType> = {
```

Reply:
Definitely not needed, and probably an outdated pattern - I just remember `I` was sometimes used to signal interface and `T` was sometimes used to signal Type. Sorry I didn't see your review before merging, will look into this more after the button stuff.